extensions/typescript-language-features/src/languageFeatures/tsconfig.ts

```diff
@@ -16,44 +18,61 @@ function mapChildren<R>(node: jsonc.Node | undefined, f: (x: jsonc.Node) => R):
 
 class TsconfigLinkProvider implements vscode.DocumentLinkProvider {
 
-	public provideDocumentLinks(
+	public async provideDocumentLinks(
 		document: vscode.TextDocument,
 		_token: vscode.CancellationToken
-	): vscode.DocumentLink[] {
+	): Promise<vscode.DocumentLink[]> {
 		const root = jsonc.parseTree(document.getText());
 		if (!root) {
 			return [];
 		}
 
 		return coalesce([
-			this.getExtendsLink(document, root),
+			await this.getExtendsLink(document, root),
 			...this.getFilesLinks(document, root),
 			...this.getReferencesLinks(document, root)
 		]);
 	}
 
-	private getExtendsLink(document: vscode.TextDocument, root: jsonc.Node): vscode.DocumentLink | undefined {
+	// Behavior reference: https://github.com/microsoft/TypeScript/blob/febfd442cdba343771f478cf433b0892f213ad2f/src/compiler/commandLineParser.ts#L3005
+	private async getExtendsLink(document: vscode.TextDocument, root: jsonc.Node): Promise<vscode.DocumentLink | undefined> {
 		const extendsNode = jsonc.findNodeAtLocation(root, ['extends']);
 		if (!this.isPathValue(extendsNode)) {
 			return undefined;
 		}
+		const extendsValue = extendsNode.value;
 
-		if (extendsNode.value.startsWith('.')) {
-			return new vscode.DocumentLink(
-				this.getRange(document, extendsNode),
-				vscode.Uri.file(join(dirname(document.uri.fsPath), extendsNode.value + (extendsNode.value.endsWith('.json') ? '' : '.json')))
-			);
-		}
+		const tsconfigUri = await this.getTsconfigPath(Utils.dirname(document.uri), extendsValue);
```

Comment:
With this PR, you are potentially performing a lot of file accesses every time the user types in the file. That's not something we can do

The current implementation also lets you create the file if it does not exist when you click the link

Reply:
Okay, in case of relative path we'll always return `.json` path in case if extensionless variant doesn't exist.

But what to do in case if `resolveNodeModulesPath()` returns `undefined`? The previous implementation would just open some path at project's `node_modules`. I don't think user want to create tsconfig at `node_modules`.

I think it makes sense to display error notification to let developer know it. (for example he just forgot to run install command)

Reply:
Also I remember when I was testing it, the link was updating the a HUGE latency. I believe that provider is already called with some latency.

Look, this behavior feels natural: IDE gives a link only when it actually finds a reference, something that points to an actual file.

IMO Creating files via non-existing link is incorrect (I understand that somebody most probably got used to it). There is should be a codefix for appropriate diagnostic instead